### PR TITLE
When giving back a judging, only give back runs from the current judg…

### PR DESCRIPTION
**Pull request by nickygerritsen**
…ehost that are not finished yet.

### CODE REVIEW

File: webapp/src/Controller/API/JudgehostController.php [hunk 1]
```diff
@@ -812,37 +810,47 @@ public function internalErrorAction(Request $request)
 
         if (in_array($disabled['kind'], ['problem', 'language', 'judgehost']) && $judgingId) {
             // Give back judging if we have to.
-            $this->giveBackJudging((int)$judgingId);
+            $hostname = $request->request->get('hostname');
+            /** @var Judgehost $judgehost */
+            $judgehost = $this->em->createQueryBuilder()
```
Comment (meisterT) [inline webapp/src/Controller/API/JudgehostController.php:815]: why not use `getRepository->find` here?

File: webapp/src/Controller/API/JudgehostController.php [hunk 2]
```diff
@@ -812,37 +810,47 @@ public function internalErrorAction(Request $request)
...
+                    // For the other runs, we need to reset the judge task if it belongs to the current judgehost
+                    if ($run->getJudgetask()->getHostname() === $judgehost->getHostname()) {
+                        $run->getJudgetask()->setHostname(null);
+                    }
+                }
+
+                $this->em->flush();
             });
 
             $this->dj->auditlog('judging', $judgingId, 'given back', null,
```
Comment (meisterT) [inline webapp/src/Controller/API/JudgehostController.php:856]: should we also update this?

File: webapp/src/Controller/API/JudgehostController.php (hunk 1 above)
Comment (nickygerritsen) [reply webapp/src/Controller/API/JudgehostController.php:815]: Copy paste from somewhere else, that sounds better

File: webapp/src/Controller/API/JudgehostController.php (hunk 2 above)
Comment (nickygerritsen) [reply webapp/src/Controller/API/JudgehostController.php:856]: What do you propose?

Comment (meisterT) [reply webapp/src/Controller/API/JudgehostController.php:856]: Perhaps mention the judge host in the message?

**Review by meisterT — APPROVED** [review]

**Pull request merged**